### PR TITLE
CUMULUS-2645 - Remove dynamodb files table/associated logic

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-2645](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2645)

## Changes

- **CUMULUS-2645**
    - Removed dynamo strctural migrations and related code from `@cumulus/api`
    - Removed unused index functionality for all tables other than
      `ReconciliationReportsTable` from `dbIndexer` lambda
    - Removed `executeMigrations` lambda
    - Removed `granuleFilesCacheUpdater` lambda
    - Removes dynamo files table from `data-persistence` module.  *This table and
      all of it's data will be removed on deployment*.
## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Changed
 
 - **CUMULUS-2312** - RDS Migration Epic Phase 3
+  - **CUMULUS-2645**
+    - Removed dynamo structural migrations and related code from `@cumulus/api`
+    - Removed unused index functionality for all tables other than
+      `ReconciliationReportsTable` from `dbIndexer` lambda
+    - Removed `executeMigrations` lambda
+    - Removed `granuleFilesCacheUpdater` lambda
+    - Removes dynamo files table from `data-persistence` module.  *This table and
```
Comment (botanical) [inline CHANGELOG.md:18]:
```suggestion
    - Removed dynamo files table from `data-persistence` module.  *This table and
```
This is a nit but the suggestion is to maintain consistency with the other changelog notes

File: packages/api/tests/lambdas/test-db-indexer.js [hunk 2]
```diff
@@ -224,196 +129,43 @@ test('getTableName() returns the full name of the DynamoDB table associated with
 
 test('getTableIndexDetails() returns undefined for unsupported table', (t) => {
   t.is(getTableIndexDetails('fake-table-name'), undefined);
-  t.is(getTableIndexDetails(GranuleFilesCache.cacheTableName()), undefined);
+  t.is(getTableIndexDetails('foobar'), undefined);
```
Comment (botanical) [inline packages/api/tests/lambdas/test-db-indexer.js:132]: We technically don't need a second verification of `getTableIndexDetails`, right? The assertion on line 131 seems sufficient?

File: tf-modules/archive/db_indexer.tf [hunk 3]
```diff
@@ -25,7 +25,6 @@ resource "aws_lambda_function" "db_indexer" {
       CollectionsTable           = var.dynamo_tables.collections.name
       ExecutionsTable            = var.dynamo_tables.executions.name
       AsyncOperationsTable       = var.dynamo_tables.async_operations.name
-      FilesTable                 = var.dynamo_tables.files.name
```
Comment (botanical) [inline tf-modules/archive/db_indexer.tf:28]: Are the variables for granules table and executions table still necessary?

**Review by botanical — COMMENTED** [review]
I believe there are two tests in `packages/api/tests/migrations` that can be removed as well

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+      all of it's data will be removed on deployment*.
```
Comment (botanical) [inline CHANGELOG.md:19]:
```suggestion
      all of its data will be removed on deployment*.
```

**Review by botanical — APPROVED** [review]
Only thing here to to remove the remaining migration tests. Otherwise, this work is valid. Thanks for handling the removal of other logic related to the files table in addition to removing the granule files cache logic 👍

File: tf-modules/archive/db_indexer.tf (hunk 3 above)
Comment (Jkovarik) [reply tf-modules/archive/db_indexer.tf:28]: Good question - they may not be.    I'm adding tickets to address table removal and/or model removal, perhaps this should go there?

File: packages/api/tests/lambdas/test-db-indexer.js (hunk 2 above)
Comment (Jkovarik) [reply packages/api/tests/lambdas/test-db-indexer.js:132]: 322ee8324d

Comment (Jkovarik) [reply packages/api/tests/lambdas/test-db-indexer.js]: 322ee8324d

**Pull request merged**